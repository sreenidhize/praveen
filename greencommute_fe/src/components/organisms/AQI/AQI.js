import { Typography, makeStyles, Grid } from "@material-ui/core";
import React from "react";
import PropTypes from "prop-types";
import Counter from "../../molecules/Counter/Counter";

const useStyle = makeStyles((theme) => ({
  descClass: {
    fontSize: "20px",
    fontWeight: 600,
    fontStretch: "normal",
    fontStyle: "nomral",
    lineHeight: 1.23,
    letterSpacing: "0.2px",
  },
  gridClass: {},
  gridItem: {},
  textClass: {
    width: "25rem",
    textAlign: "center",
    fontSize: "20px",
    fontWeight: 600,
  },
  margin: {
    marginTop: "3%",
  },
}));

function AQI(props) {
  const { image, counters, description, ...rest } = props;
  const classes = useStyle();

  return (
    <Grid
      container
      direction="column"
      spacing={6}
      alignItems="center"
      justify="center"
      className={classes.gridClass}
    >
      <Grid item xs={12} className={classes.gridItem} align="center">
        <img alt="map" src={image} {...rest} />
      </Grid>
      <Grid
        item
        container
        xs={12}
        alignItems="center"
        justify="center"
        className={classes.margin}
        // spacing={4}
      >
        {counters &&
          counters.map((aqi, index) => (
            <Grid item xs={5} key={index}>
              <Counter
                city={aqi.city}
                number={aqi.number}
                showCity={aqi.showCity}
              />
            </Grid>
          ))}
      </Grid>
      <Grid item xs={12}>
        <Typography variant="subtitle1" className={classes.textClass}>
          {description}
        </Typography>
      </Grid>
    </Grid>
  );
}

AQI.propTypes = {
  image: PropTypes.string,
  counters: PropTypes.array,
  description: PropTypes.string,
};

AQI.defaultProps = {
  counters: [],
};

export default AQI;
