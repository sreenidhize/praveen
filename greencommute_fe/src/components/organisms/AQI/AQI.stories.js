import React from "react";
import AQI from "./AQI";
import imageMap1 from "../../../assets/map1.png";
import imageMap2 from "../../../assets/map2.png";
import imageMap3 from "../../../assets/map3.png";

const AQIStory = {
  title: "Organisms/AQI",
  component: AQI,
};

const Template = (args) => <AQI {...args} />;

export const LandingVariation1 = Template.bind({});

LandingVariation1.args = {
  image: imageMap1,
  description: "Enter location to know Time Air Quality Index (AQI)",
};

export const LandingVariation1a = Template.bind({});

LandingVariation1a.args = {
  image: imageMap1,
  counters: [{ number: "894", showCity: false }],
  description: "Real - Time Air Quality Index (AQI) in this location",
};

export const LandingVariation2 = Template.bind({});

LandingVariation2.args = {
  image: imageMap2,
  description: "Enter location to know Time Air Quality Index (AQI)",
};

export const LandingVariation2a = Template.bind({});

LandingVariation2a.args = {
  image: imageMap2,
  counters: [
    { number: "894", showCity: true, city: "Hyderabad" },
    { number: "953", showCity: true, city: "Mumbai" },
  ],
  description: "Real - Time Air Quality Index (AQI) in this location",
};

export const LandingVariation3 = Template.bind({});

LandingVariation3.args = {
  image: imageMap3,
  description: "Enter your skills to know how many jobs are in this location",
};

export const LandingVariation3a = Template.bind({});

LandingVariation3a.args = {
  image: imageMap3,
  counters: [{ number: "953", showCity: false }],
  description: "Jobs found in these locations",
};

export default AQIStory;
