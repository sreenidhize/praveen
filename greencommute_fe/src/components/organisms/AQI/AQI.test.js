import React from "react";
import AQI from "./AQI";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("AQI renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<AQI />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should reder Image", () => {
    render(
      <AQI
        image={"Dummmy"}
        description="Enter location to know Time Air Quality Index (AQI)"
        data-testid="AQI"
      />
    );
    screen.getAllByTestId("AQI");
  });
});
