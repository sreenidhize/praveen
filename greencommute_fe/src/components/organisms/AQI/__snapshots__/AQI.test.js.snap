// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AQI renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-gridClass-2 MuiGrid-container MuiGrid-spacing-xs-6 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      align="center"
      class="MuiGrid-root makeStyles-gridItem-3 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <img
        alt="map"
      />
    </div>
    <div
      class="MuiGrid-root makeStyles-margin-5 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
    />
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h6
        class="MuiTypography-root makeStyles-textClass-4 MuiTypography-subtitle1"
      />
    </div>
  </div>
</DocumentFragment>
`;
